infracheck/infracheck/model.py

```diff
@@ -1,5 +1,108 @@
-from datetime import datetime
-from typing import Dict
+from datetime import datetime, timedelta
+from typing import Dict, List, Union, Optional
+from dataclasses import dataclass
+from croniter import croniter
+from rkd.api.inputoutput import IO
+
+from infracheck.infracheck.exceptions import ConfigurationException
+
+QUIET_PERIODS_DATA_STRUCT = List[Dict[str, Union[str, int]]]
+HOOKS_STRUCT = Dict[str, List[str]]
+INPUT_VARIABLES_STRUCT = Dict[str, Union[str, int]]
+
+
+@dataclass
+class ConfiguredCheck(object):
+    name: str
+    check_type: str  # type
+    description: str
+    input_variables: INPUT_VARIABLES_STRUCT
+    hooks: HOOKS_STRUCT
+    quiet_periods: QUIET_PERIODS_DATA_STRUCT
+    results_cache_time: Optional[int]
+    io: IO
+
+    @classmethod
+    def from_config(cls, name: str, config: dict, io: IO):
+        quiet_periods = config.get('quiet_periods', [])
+
+        if not isinstance(quiet_periods, list):
```

Comment:
In Python there is no such strict type checking unfortunately, so it can be anything - but at first level it must be a list.
If you wish there could be a separate ticket created to use ex. JSON Schema to validate such thing.